packages/netstring/writer.js

```diff
@@ -1,51 +1,54 @@
 // @ts-check
 /// <reference types="ses"/>
 
-const COLON = ':'.charCodeAt(0);
-const COMMA = ','.charCodeAt(0);
+const COMMA_BUFFER = new Uint8Array([','.charCodeAt(0)]);
 
-// The initial buffer length should be small enough to not be an imposition in
-// the common case but large enough to avoid reallocation in the common case.
-// It also must be long enough to fit an ASCII decimal representation of the
-// length of the longest possible message.
-// This is a guess:
-const initialBufferLength = 128;
-
-const encoder = new TextEncoder();
+/** @param {number} length */
+const getLengthPrefixCharCodes = length =>
+  // eslint-disable-next-line no-bitwise
+  [...`${length | 0}:`].map(char => char.charCodeAt(0));
 
 /**
  * @param {import('@endo/stream').Writer<Uint8Array, undefined>} output
- * @returns {import('@endo/stream').Writer<Uint8Array, undefined>}
+ * @param {object} [opts]
+ * @param {boolean} [opts.chunked]
+ * @returns {import('@endo/stream').Writer<Uint8Array | Uint8Array[], undefined>}
  */
-export const makeNetstringWriter = output => {
+export const makeNetstringWriter = (output, { chunked = false } = {}) => {
   return harden({
-    async next(message) {
-      // Must allocate to support concurrent writes.
-      let buffer = new Uint8Array(initialBufferLength);
-      const { written: colonAt = 0 } = encoder.encodeInto(
-        `${message.byteLength}`,
-        buffer,
+    async next(messageChunks) {
+      if (!Array.isArray(messageChunks)) {
+        messageChunks = [messageChunks];
+      }
+
+      const messageLength = messageChunks.reduce(
+        (acc, { length }) => acc + length,
+        0,
       );
```

Comment:
This could have been a for loop.

Reply:
A simple accumulator is the one case where I feel like `reduce` is acceptable but if it hurts too much to read, I'm happy to make it a loop.